R/vars_manager.R

```diff
@@ -243,151 +281,176 @@ get_dataframe_var <- function(
         } else {
-            df_sort <- data.frame(
-                col = sort_by,
-                asc = TRUE
-            )
+            obj_pre <- obj_filtered[start:end,]
         }
 
-        for (row in 1:nrow(df_sort)) {
-            params <- append(params, convert_param_to_character(df_sort[row,]))
-        }
-
-        # Sort by params generated above.
-        # 'arrange' discards row.names, so we need to preserve them manually
-        obj_sorted <- obj_filtered %>%
-                        rownames_to_column('_callisto_r_rows_') %>%
-                        arrange(!!! parse_exprs(params)) %>%
-                        column_to_rownames('_callisto_r_rows_')
-        obj_pre <- `if`(abbrev, obj_sorted[1:abbrev_len,], obj_sorted)
-    } else {
-        obj_pre <- `if`(abbrev, obj_filtered[1:abbrev_len,], obj_filtered)
-    }
-
-    data <- list()
-    if (length(obj_pre) > 0) {
-        for (i in 1:nrow(obj_pre)) {
-            # If we use as.character on the whole row at once factors are represented as 
-            # numbers and not their text value, so we as.character each cell
-            data_row = list()
-            for (j in 1:ncol(obj_pre)) {
-                data_row[[j]] = as.character(obj_pre[i,j])
+        data <- list()
+        if (length(obj_pre) > 0) {
+            for (i in 1:nrow(obj_pre)) {
+                # If we use as.character on the whole row at once factors are represented as 
+                # numbers and not their text value, so we as.character each cell
+                data_row = list()
+                for (j in 1:ncol(obj_pre)) {
+                    data_row[[j]] = as.character(obj_pre[i,j])
+                }
+                data[[i]] = data_row
             }
-            data[[i]] = data_row
         }
-    }
-    col_names <- list()
-    if (ncol(obj_pre) > 0) {
-        i <- 1
-        for (col_name in `if`(length(names(obj_pre)) > 1, names(obj_pre), 1:ncol(obj_pre))) {
-            col_names[i] <- sprintf("%s (%s)", as.character(col_name), toString(class(obj_pre[,i])))
-            i <- i + 1
+        col_names <- list()
+        if (ncol(obj_pre) > 0) {
+            i <- 1
+            for (col_name in `if`(length(names(obj_pre)) > 1, names(obj_pre), 1:ncol(obj_pre))) {
+                col_names[i] <- sprintf("%s (%s)", as.character(col_name), toString(class(obj_pre[,i])))
+                i <- i + 1
+            }
         }
-    }
-    row_names <- list()
-    if (nrow(obj_pre) > 0) {
-        i <- 1
-        for (row_name in `if`(length(rownames(obj_pre)) > 1, rownames(obj_pre), 1:nrow(obj_pre))) {
-            row_names[i] <- as.character(row_name)
-            i <- i + 1
+        row_names <- list()
+        if (nrow(obj_pre) > 0) {
+            i <- 1
+            for (row_name in `if`(length(rownames(obj_pre)) > 1, rownames(obj_pre), 1:nrow(obj_pre))) {
+                row_names[i] <- as.character(row_name)
+                i <- i + 1
+            }
```

Comment:
same as above, if there's no specific rownames then the row_names should reflect the page you're on